website/docs/language/block/output.mdx

```diff
@@ -0,0 +1,289 @@
+
+```hcl
+output "unique_name" {
+  value      = <expression-to-evaluate>
+  depends_on = [<another_resource.name>]
+}
+```
+
+If the `value` expression in an `output` block references another resource, Terraform automatically determines the dependency order. If you need to establish a dependency between resources and outputs that are configured independently and do not reference each other, you can use the`depends_on` argument to declare an explicit dependency. If you add the `depends_on` argument, we recommend adding a comment to explain it.
+
+The `depends_on` block is a meta-argument. Meta-arguments are built-in arguments that control how Terraform creates resources. Refer to [Meta-arguments](/terraform/language/meta-arguments/depends_on) for additional information.
+
+<>{/* TODO: Update meta-arguments concept link once we have the page*/}</>
+
+#### Summary
+
+- Data type: List of references
+- Default: None
+
+### `precondition`
+
+The `precondition` block lets you validate that the value of an `output` block meets specific requirements. Use preconditions on outputs to validate that your output values meet your requirements before Terraform exposes them or stores their values in state.
+
+```hcl
+output "unique_name" {
+  value  = <expression-to-evaluate>
+
+  precondition {
+    condition     = <expression-to-verify>
+    error_message = "<error message string>"
+  }
+}
+```
+
+You can specify the following arguments in the `precondition` block:
+
+| Attribute | Description | Data type | Required? |
+| :---- | :---- | :---- | :---- |
+| `condition` | Expression that must return `true` for Terraform to proceed with an operation. | A valid [Conditional expression](/terraform/language/expressions/conditionals). | Required |
+| `error_message` | Message to display if the condition evaluates to `false`. | String | Required |
+
+Terraform evaluates preconditions on outputs when creating or applying a plan, and if a precondition block's `condition` expression evaluates to `false`, then Terraform throws an error with the `error_message` and stops the current operation.
+
+Refer to [Validate your configuration](/terraform/language/test-and-validate/validate#preconditions) for more details about preconditions.
+
+#### Summary
+
+- Data type: Block
+- Default: None
+
+## Examples
+
+The following examples demonstrate common use cases for `output` values.
+
+### Basic example
+
+In the following example, an `output` block exposes a server's private IP address from the current module:
+
+```hcl
+output "instance_ip_addr" {
+  value       = aws_instance.server.private_ip
+  description = "The private IP address of the main server instance."
+}
+```
+
+### Accessing child module outputs
+
+In the following example, the current module uses an output named `instance_ip_addr` from a child module named `web_server`:
+
+```hcl
+output "website_url" {
+  value = "https://${module.web_server.instance_ip_addr}"
+}
+```
```

Comment:
As best practice, I think all of these should have a description field.
description = "The website URL, starting with https://"